php/context/class-fieldmanager-context-storable.php

```diff
@@ -52,9 +52,18 @@ protected function save( $data = null ) {
 			$this->save_field( $this->fm, $data, $this->fm->data_id );
 		} else {
 			if ( null === $data ) {
-				$data = isset( $_POST[ $this->fm->name ] ) ? $_POST[ $this->fm->name ] : ''; // WPCS: input var okay. CSRF okay. Sanitization okay.
+				$data = isset( $_POST[ $this->fm->name ] ) ? $_POST[ $this->fm->name ] : array(); // WPCS: input var okay. CSRF okay. Sanitization okay.
+
+				// Multiple select and radio fields with no values chosen are left out of the post request altogether.
+				foreach ( $this->fm->children as $child ) {
+					$is_multi_select = ( 'select' == $child->field_class ) && ( true === $child->multiple );
+					$is_radio = ( 'radio' == $child->field_class );
```

Comment:
Perhaps this could use `instanceof` instead of the `field_class` property, which is public. See 878f9e55a19882444eb0a4fa0f8108e192a6a89f.